File: 3832-minimum-cost-to-reach-every-position/minimum-cost-to-reach-every-position.cpp
class Solution {
public:
vector<int> minCosts(vector<int>& cost) {
    int minsfr = INT_MAX;
    int n=cost.size();
    for (int i = 0; i < n; i++) {
        minsfr = min(minsfr, cost[i]);
        cost[i] = minsfr; 
    }
    return cost;
}
};

File: 3832-minimum-cost-to-reach-every-position/minimum-cost-to-reach-every-position.py
class Solution:
    def minCosts(self, cost: List[int]) -> List[int]:
        n = len(cost)
        minBefore = cost[:]
        min_val = cost[0]
        for i in range(1, n):
            minBefore[i] = min(minBefore[i], minBefore[i-1])
        
        return minBefore